### ISSUE TITLE
[android] Position arrow overlaps with bottom bar in navigation mode.

### ISSUE THREAD

**Issue by vng** [organicmaps/organicmaps#3441]
Samsung Galaxy S20 FE (Snapdragon), latest beta:

Upd: looks like it happens only when device is locked

![photo_2022-09-18 12 08 19](https://user-images.githubusercontent.com/175612/190894610-a2285f11-d447-4e30-b919-968863cc41c4.jpeg)

**Comment by arnaudvergnet** [comment]
Is the font size changed in Android settings? This may have been introduced by https://github.com/organicmaps/organicmaps/pull/3085. Before this patch the navigation bar was of fixed height (so text could be cut off if the font was too large). Now the height scales with the content so if the font is too large, this issue may occur.

A good fix would be to move the camera depending on the navigation menu height but I have no clue on how to do that.

**Comment by biodranik** [comment]
@vng let's slightly increase the arrow height first. Even without this issue, it was positioned too low.

@arnaudvergnet A proper fix should take into account the nav bar, of course. So we have several choices here:
1. Pass the new height into the core so it can adjust the arrow height. Can be tricky, and may lead to an unusable UI if a user sets a very large font and has a very small screen.
2. Fix the height of the nav bar as it was before, but dynamically format/reduce the font size in the displayed info to fit it. That is easier to implement, but can be inconvenient for some users with bad eyesight. Can be a good temporary solution though.
3. Get rid of the nav bar completely (or partially by making a hole in the center) in favor of a new UI/UX, as we did for the main map screen. The current speed and speed limit can be displayed in a circle somewhere in the (top?) corner. The remaining time and distance can also be displayed somewhere else (e.g. in the bottom left and bottom right corners, or even in the top street info bar).

**Comment by rtsisyk** [comment]
This bug still present in the current master. On Samsung S10 (Android 12) the arrow is not overlapped, but it is noticeably lower than it was before. On Pixel 4a (Android 13) the ends of the arrow are overlapped a little bit.

**Comment by biodranik** [comment]
It is resolved, right?

**Comment by biodranik** [comment]
Some users complain that the arrow is too low.

**Comment by biodranik** [comment]
One more example. Note that the layout of the top bar should be also fixed @Jean-BaptisteC 

https://user-images.githubusercontent.com/170263/225320416-6837eb72-18d5-4be6-af41-e0930b31cca9.mp4

**Comment by biodranik** [comment]
@rtsisyk can #3462 fix it?

**Comment by rtsisyk** [comment]
Toyak Alak in Telegram chat [complained about this issue a couple of days ago](https://t.me/organicmaps/30760).

> S20 FE, stock font size, normal landscape
>> In which version did arrow work for you?
> A very old versions (before May 2023)

https://github.com/organicmaps/organicmaps/assets/1799054/9a549dd4-e218-4c6f-be10-e665a8f0afaa

**Comment by rtsisyk** [comment]
I believe that I found that race condition that caused this inconsistency in #6556. Please don't hesitate to re-open if problem continues to occur after 2023.11.19-3 (should be in the next release).

**Comment by biodranik** [comment]
@rtsisyk  The closed #6923 issue had important steps to reproduce the bug:

Android 14 device with Vulkan renderer (doesn't reproduce in Emulator with OpenGL).

1. Start navigation mode (preferably cycling or walking, reproduces easier than in driving).
2. Lock the phone with power button.
3. Unlock the phone in a few seconds.
4. Observe that location arrow has moved under the bottom bar.
5. Lock the phone again.
6. Unlock in a few seconds.
7. Observe that location arrow has moved back at its position

**Comment by rutschi100** [comment]
I have exactly the same problem. I mainly use it for navigation on foot. This means that the intersection I am at is not visible and I have to move the view manually. The problem with this is that the automatic alignment is lost. So I have the choice of using the navigation reasonably well or seeing nothing at all but having automatic alignment.

Will the positioning be addressed soon? PS: I haven't changed the font size or anything like that, everything is default.

[logs.zip](https://github.com/user-attachments/files/24991407/logs.zip)

**Comment by biodranik** [comment]
@rutschi100 does it look the same as on the initial screenshot?

**Comment by rutschi100** [comment]
@biodranik Yes, exactly the same. I just noticed another behaviour that helps reproduce the issue. Initially, when starting the route, it looks normal. As soon as I lock the mobile phone and leave only the screen on without unlocking it, it happens. When I unlock it, it looks normal again. So the problem only occurs when my mobile phone is locked.

![Image](https://github.com/user-attachments/assets/b4390442-00d2-489f-9871-48aaf2e5058f)

**Comment by biodranik** [comment]
Thanks for confirming! Does your device have software buttons?

Looks like the arrow position formula doesn't take some insets specifically on the lock screen (e.g. missing system buttons, or something else). 


An obvious workaround is to disable OM on a lock screen in OM settings.
